lib/data_source/client/github.rb

```diff
@@ -29,30 +42,86 @@ def user_repos
           direction: 'desc',
           per_page: 100
         }
-        with_cache('user_repos') do
-          request('/user/repos', params)
+        responses = with_cache(:user_repos) do
+          all_user_repos = merge_multipage_results('/user/repos', params, 100)
+          all_user_repos
+        end
+
+        all_user_repos = Array.new
+        responses.each do |response|
+          all_user_repos = all_user_repos + response
         end
+        all_user_repos
       end
 
       def user_pulls
-        modifiers = org_modifiers('is:pr', 'user:@me', 'state:open', 'involves:@me')
+        if @pr_all_involve_me
+          modifiers = ['is:pr', 'state:open', "involves:#@me_account"]
+        else
+          modifiers = org_modifiers('is:pr', "user:#@me_account", 'state:open', "involves:#@me_account")
+        end
         params = search_params('', modifiers).merge(
           per_page: 100
         )
-        response = with_cache('user_pulls') do
-          request('/search/issues', params)
+        responses = with_cache(:user_pulls) do
+          all_user_pulls = merge_multipage_results('/search/issues', params, 100)
+          all_user_pulls
+        end
+        all_user_pulls = Array.new
+        responses.each do |response|
+          all_user_pulls = all_user_pulls + response[:items]
         end
-        response[:items]
+        all_user_pulls
       end
 
       private
 
+      def merge_multipage_results(path, params, per_page)
+        params[:per_page] = per_page
+        params[:page] = 1
+        res = request(path, params)
+        raise res[:message] unless res.is_a?(Net::HTTPSuccess)
+
+        result = Array.new
+        result.append deserialize_body(res.body)
+
+
+        page_count = 1
+        if res.key?("Link")
+          res["Link"].split(",").map do |result|
+            page_num, rel = result.match(/&page=(\d+)>; .*?"(\w+)"/i).captures
+            page_count = page_num.to_i if rel == "last"
+          end
+        end
```

Comment:
This is a clever solution! Thank you. I'll just add my two cents: Instead of doing a `map`, I would use `find`, so that when the item you're looking is found, it stops the iteration. Although is possible that the `rel="last"` item is the last one, in which case maybe it's worth doing a reverse: `res['Link'].split(',').reverse.find do |result|`.

And also I would move this logic to another method called `results_last_page(res)`